blog/uncertainty_quantification_deep_learning_understanding.md

```diff
@@ -0,0 +1,127 @@
+    * Normal is likelihood function
+    * $\mu$, $\sigma^2$ are distribution parameters
+
+|  | Classification | Regression |
+|---|---|---|
+|Targets| y $\in$ {1,...,k} | y $\in$ $\mathbb{R}$ | 
+|Likelihood | y ~ Categorical(p) | y ~ Normal( $\mu$, $\sigma^2$) |
+|Parameters | p = {p1,...,pk} | ( $\mu$, $\sigma^2$)|
+|Constraints | $\sum_{i} p_{i}$ = 1;  $p_{i}$ > 0| $\mu \in \mathbb{R}$; $\sigma$ > 0|
+|Loss function | Cross Entropy $$-\sum_{i=1}^{K} y_{i} logp_{i}$$ | Negtaive Log-Likelihood $$-log(N(y \| \mu, \sigma^2))$$ |
+
+The drawbacks of likelihood is it's unreliable outputs, if the input is unlike anything during training. <br>
+Example: Feeding a ship image in a Cat, Dog trained model. Though the probabilities of likelihood be anything, it cannot be trusted and confidence value is required. This is beacuse of the uncertainty and their types are:
+
+| Aleatoric | Epistemic |
+|---|---|
+|Data uncertainty| Model uncertainty |
+|Describes the confidence in input data | Describes the confidence of prediction |
+|High when input data is noisy | High when missing training data | 
+|Cannot be reduced by adding more data | Can be reduced by adding more data| 
+|Can be learned directly using NN with likelihood estimation techniques | Can be solved using Bayesian NN instead of deterministic NN| 
+|For regression problem, giving data x to model with likelihood estimation technique provides ( $\mu$, $\sigma^2$)| For the same problem, giving data x to model with likelihood estimation techniques and ensemble (many independently trained instances of the model), multiple ( $\mu$, $\sigma^2$) are produced|
+|$\sigma^2$ is the data uncertainty | The variance of $\mu$ from multiple instances of ensemble is model uncertainty. If var[ $\mu$] is lesser, the model is more confident and vice versa| 
+
+| Distribution parameter | Type of uncertainty|
+|---|---|
+|Low variance in $\mu$ and $\sigma^2$| Low uncertainty |
+|High variance in $\sigma^2$| Data uncertainty|
+|High variance in $\mu$ | Model uncertainty|
+
+
+| Deterministic NN | Bayesian NN |
+|---|---|
+|Single number representation for every weight| Probability distribution representation for every weight|
+|Learn fixed set of weights $$W$$ | Learn posterior over weights $$P(W\|X,Y)$$ Learns posterior with sampling methods such as dropout, ensembles|
+|Given a set of weights, passing in one input to the model multiple times will yield the same fixed output | sampling of the weight distribution is performed for every single weight resulting in slightly different outputs for different iterations|
+
+
+# Evidential Deep learning
+Instead of sampling to learn posteriors such as in ensemble methods, the distribution of the likelihood parameters can be learnt directly and are called as evidential deep learning. This distribution is a higher order distribution of the likelihood distributions parameters and evidential deep learning directly estimates both aleatoric and epistemic uncertainty.
+
```

Comment:
I think the correct statement is 
Instead of sampling in ensembles and dropout and estimating the posterior, we can directly learn the posterior distribution parameter.  This idea was taken by Nix, D. A. and Weigend, A. S., “Estimating the mean and variance of the target probability distribution”, and modeled the output as a gaussian distribution with 2 output mu and sigma . Deebul updated it the laplace distribution to check robustness to outliers .
In evidential deep learning the output is modeled as higher order distirbution  like Normal-Inverse-Gamma or Dirichlet distirbutions.